homeassistant/components/jellyfin/media_player.py

```diff
@@ -0,0 +1,463 @@
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Jellyfin media_player from a config entry."""
+    client: JellyfinClient = hass.data[DOMAIN][config_entry.entry_id][DATA_CLIENT]
+    user = await hass.async_add_executor_job(client.jellyfin.get_user)
+
+    coord = JellyfinMediaPlayerCoordinator(hass, client, user, async_add_entities)
+    await coord.async_config_entry_first_refresh()
+    _LOGGER.debug("New entity listener created")
+
+
+class JellyfinMediaPlayerCoordinator(DataUpdateCoordinator):
+    """Bundles data retrieval for sessions form Jellyfin server."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        client: JellyfinClient,
+        user: str,
+        async_add_entities: AddEntitiesCallback,
+    ) -> None:
+        """Initialize the UpdateCoordinator."""
+        super().__init__(
+            hass, _LOGGER, name="Jellyfin", update_interval=timedelta(seconds=5)
+        )
+
+        self._client = client
+        self._hass = hass
+        self.user = user
+        self.sessions: Any | None = None
+
+        self._known: set[str] = set()
+        self._async_add_entities = async_add_entities
+
+    async def _async_update_data(self) -> None:
+        self.sessions = await self._hass.async_add_executor_job(
+            self._client.jellyfin.get_sessions
+        )
+
+        if not self.sessions:
+            return
+
+        for session in self.sessions:
+            if session["Id"] not in self._known:
+                self._known.add(session["Id"])
+                entity = JellyfinMediaPlayer(self, self._hass, self._client, session)
+                self._async_add_entities([entity], False)
+
+
+class JellyfinMediaPlayer(CoordinatorEntity, MediaPlayerEntity):
+    """Represents a Jellyfin Player device."""
+
+    def __init__(
+        self,
+        coordinator: JellyfinMediaPlayerCoordinator,
+        hass: HomeAssistant,
+        client: JellyfinClient,
+        session: Any,
+    ) -> None:
+        """Initialize the MediaPlayer Entity.
+
+        Pulls out the properties from the session object we need, even when unavailable.
+        """
+        super().__init__(coordinator)
+        self._hass = hass
+        self._client = client
+        self._device_id = session["DeviceId"]
+        self._id = session["Id"]
+
+        self._data = session
+
+        self._attr_unique_id = f"{self._data['ServerId']}:{self._id}"
+        self._attr_name = self._data["DeviceName"]
+        self._attr_media_position_updated_at = homeassistant.util.dt.utcnow()
+        self._attr_media_content_type = "video"
+        self._attr_media_image_remotely_accessible = True
+        self._attr_should_poll = True
+
+    def update(self) -> None:
+        """Update the internal state from Coordinator state."""
+        self._data = None
+        for session in self.coordinator.sessions:
```

Comment:
for future might consider indexing the sessions by id so you don't need to loop in the media player